File: snippets/fsharp/System/String/Join/join5.fs
module join5.fs
// <Snippet5>
// This example uses F#'s Seq.filter function instead of Linq.
open System

type Animal =
  { Kind: string
    Order: string }
    override this.ToString() =
        this.Kind

let animals = ResizeArray()
animals.Add { Kind = "Squirrel"; Order = "Rodent" }
animals.Add { Kind = "Gray Wolf"; Order = "Carnivora" }
animals.Add { Kind = "Capybara"; Order = "Rodent" }
String.Join(" ", animals |> Seq.filter (fun animal -> animal.Order = "Rodent"))
|> printfn "%s"
// The example displays the following output:
//      Squirrel Capybara
// </Snippet5>

File: snippets/csharp/System/String/Join/join5.cs
// <Snippet5>
using System;
using System.Collections.Generic;
using System.Linq;

public class Animal
{
    public string Kind;
    public string Order;

    public Animal(string kind, string order)
    {
        this.Kind = kind;
        this.Order = order;
    }

    public override string ToString() => this.Kind;
}

public class Example
{
    public static void Main()
    {
        List<Animal> animals = new()
        {
            new Animal("Squirrel", "Rodent"),
            new Animal("Gray Wolf", "Carnivora"),
            new Animal("Capybara", "Rodent")
        };
        string output = string.Join(" ", animals.Where(animal =>
                        (animal.Order == "Rodent")));
        Console.WriteLine(output);
    }
}
// The example displays the following output:
//      Squirrel Capybara
// </Snippet5>
